src/sampling.jl

```diff
@@ -590,6 +590,7 @@ Optionally specify a random number generator `rng` as the first argument
 function sample(rng::AbstractRNG, wv::AbstractWeights)
     1 == firstindex(wv) ||
         throw(ArgumentError("non 1-based arrays are not supported"))
+    isfinite(sum(wv)) || throw(ArgumentError("only finite weights are supported"))
```

Comment:
A general comment: Could we avoid re-computing `sum(wv)` below? It's not guaranteed that `sum(wv)` is cheap for any `AbstractWeights` type.

Reply:
Isn't the sparse documentation an additional argument for avoiding re-evaluating `sum(wv)` and not relying on such fast paths?